### DISCUSSION TITLE
entangle model property not a thing?

### DISCUSSION THREAD

**Discussion by torian257x** [livewire/livewire#4360] [1. Help] [answered]
hi
i've been searching around quite a bit

I can `@entangle` primitives but not model properties e.g.

```
<div x-data={userId: @entangle('user.id')}> 
...
```

is there a way to do it anyhow? maybe just bind?

**Comment by joshhanley** [comment, marked as answer]
@torian257x that should work, but you need to make sure you have a rule for `user.id` otherwise it won't be accessible by the front end and therefore alpine/entangle won't work.

See here in the docs https://laravel-livewire.com/docs/2.x/properties#binding-models
> Note: For this to work, you have a validation entry in the `$rules` property for any model attributes you want to bind to

So in your component add

```php
protected $rules [
    'user.id' => 'required', // or whatever validation you need
];
```

Hope this helps!

**Reply by torian257x** [reply]
ah, I knew about the rules, but somehow couldnt recall it when I was reading the error...

**Reply by joshhanley** [reply]
No worries! 🙂

**Reply by SamuelBoudine** [reply]
Hello. My property is a object, after the component refresh, my property convert to string, why?
